src/app/contract/verifySignatureCollisions.js

```diff
@@ -0,0 +1,23 @@
+'use strict'
+
+var request = require('xhr-request')
+
+module.exports = (methodIdentifiers) => {
+  return verifySignatureAsync(methodIdentifiers)
+}
+
+async function verifySignatureAsync (methodIdentifiers) {
+  var collisions = []
+  for (let func in methodIdentifiers) {
+    var signature = methodIdentifiers[func]
+    var url = 'https://raw.githubusercontent.com/ethereum-lists/4bytes/master/signatures/' + signature
+    request(url, (err, data, response) => {
+      if (err) {
+        console.log(err)
+      } else if (response.statusCode === 200 && (data.split(';').length > 1 || data !== func)) {
+        collisions.push(func)
+      }
+    })
+  }
+  return collisions
```

Comment:
That would not solve the original issue here, cause you are returning `collisions` before it is filled

Reply:
I'm calling this module as async in the others files, it fixed that problem.

Reply:
Is there any other way to solve that? Which do you prefer?

Reply:
The solutions I would suggest are to either return either a promise (and adjust code elsewhere to reflect that?) or to add a callback function as a param and call that function upon a collision (after `collisions.push`), however I don't know which @yann300 would prefer to fit into the style of the codebase as-is

Reply:
>  to either return either a promise

that would be fine to do like that

Reply:
Hi, sorry for the delay, my day was to busy. Tomorrow I will try to implement the your suggestion (@DanielRX ), may I come here with some questions, I'm not too good with Promises.